shared/src/native-src/loader.cpp

```diff
@@ -1391,6 +1536,125 @@ namespace shared
+
+                // We need to iterate and find the right section
+                auto section = (IMAGE_SECTION_HEADER*) pSection;
+                for (auto i = 0; i < VAL16(ntHeaders->FileHeader.NumberOfSections); i++)
+                {
+                    // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#section-data
+                    const auto sectionVirtualAddress = VAL32(section->VirtualAddress);
+                    const auto sectionVirtualSize = (UINT) VAL32(section->Misc.VirtualSize);
+
+                    // The alignment (in bytes) of sections when they are loaded into memory.
+                    // It must be greater than or equal to FileAlignment. The default is the page size for the
+                    // architecture.
+                    const auto sectionAlignment = (UINT) VAL32(ntHeaders->OptionalHeader.SectionAlignment);
+
+                    const auto sectionAlignUpValue =
+                        (sectionVirtualSize + sectionAlignment - 1) & ~(sectionAlignment - 1);
+
+                    if (clrDataVirtualAddress < sectionVirtualAddress + sectionAlignUpValue)
+                    {
+                        if (clrDataVirtualAddress < sectionVirtualAddress)
+                        {
+                            Debug("Loader::GetModuleEntryPointToken: " + ToString(i) + ". sectionRet = nullptr");
+                            sectionRet = nullptr;
+                        }
+                        else
+                        {
+                            Debug("Loader::GetModuleEntryPointToken: " + ToString(i) + ". sectionRet = (found)");
+                            sectionRet = section;
+                            break;
+                        }
+                    }
+
+                    // Advance to the next section.
+                    section++;
+                }
+
+                if (sectionRet == nullptr)
+                {
+                    pCorHeader = moduleBaseLoadAddress + clrDataVirtualAddress;
+                }
+                else
+                {
+                    pCorHeader = moduleBaseLoadAddress + clrDataVirtualAddress - VAL32(sectionRet->VirtualAddress) +
+                                 VAL32(sectionRet->PointerToRawData);
+                }
+            }
+            else
+            {
+                pCorHeader = moduleBaseLoadAddress + clrDataVirtualAddress;
+            }
+
+            // Get the COR Header
+            const auto corHeader = (IMAGE_COR20_HEADER*) pCorHeader;
+            return corHeader->EntryPointToken;
+        }
+        catch (...)
+        {
+            auto ex = std::current_exception();
+            try
+            {
+                if (ex)
+                {
+                    std::rethrow_exception(ex);
+                }
+            }
+            catch (const std::exception& ex)
+            {
+                Error("Failed getting module entrypoint: " + ToString(ex.what()));
+            }
```

Comment:
In the outer-catch, you get the current exception, then if it's not nullptr, rethrow it to catch it again and log.
instead of `catch(...)` why not `catch(const std::exception&)` ?

Can the exception really be null in that case ?


why not having just
```cpp
catch (. . .)
{
   auto what = ex ? ToString(ex.what()) : ToString("nullptr exception");
   Error("...", what);
}
```

Reply:
I copied that snipped from the [current tracer use](https://github.com/DataDog/dd-trace-dotnet/blob/master/src/Datadog.Trace.ClrProfiler.Native/integration_loader.cpp#L91-L105), I don't have strong opinion on that. I can change it if you want.

Reply:
I do not have a strong opinion, I was hoping you knew the underlying reason